src/client/datascience/interactive-window/interactiveWindow.ts

```diff
@@ -171,12 +172,13 @@ export class InteractiveWindow implements IInteractiveWindowLoadable {
     private async createKernelReadyPromise(): Promise<IKernel> {
         const editor = await this._editorReadyPromise;
         const controller = await this._controllerReadyPromise.promise;
+        initializeNotebookTelemetryBasedOnUserAction(editor.document.uri, controller!.connection);
         const kernel = this.kernelProvider.getOrCreate(editor.document, {
             metadata: controller!.connection,
             controller: controller!.controller,
             resourceUri: this.owner
         });
-        await kernel.start({ disableUI: false, document: editor.document });
+        await kernel.start({ disableUI: false });
```

Comment:
the kernel is tied to a notebook (1-1 relationship), hence moot point to pass the notebook all over again to the kernel